resolver_service/views.py

```diff
@@ -269,6 +245,34 @@ def processRequest(self):
             return self.returnResponseLinkTypeData(getURLandTitle(bibcode=self.bibcode, linkType=self.linkType, linkSubType=self.linkSubType),
                                                    self.config['RESOLVER_GATEWAY_URL'])
 
-        # we did not recognize the linkType, so return an error
-        return self.__returnResponseError("error: unrecognizable linkType:'{linkType}'!".format(linkType=self.linkType), 400)
 
+    
+    def processRequest(self):
+        
```

Comment:
i have provided an exaple of the logic - some parts are missing:

- exceptions are not defined
- _get_url method is not working

the main point is this: microservice needs to return JSON (only in very few exceptional cases can it return something else) - so for the resolver service, there is one potential special case. Directly redirect (302) to the target url

but in all other circumstances it needs to respond with data; never HTML

HTML is acceptable/expected for the gateway though

Reply:
btw: you will notice i've removed the 'referredUrl' variables - that is precisely because we are inside a microservice; this microservice doesn't need to know the referrer - it is the gateway that nees to know it

Reply:
>>you will notice i've removed the 'referredUrl' variables
thank you. it was leftover from when logging was done here, it is now moved to resolver_gateway.